Tests/SwiftIDEUtilsTest/ClassificationTests.swift

```diff
@@ -121,4 +121,379 @@ public class ClassificationTests: XCTestCase {
+        ClassificationSpec(source: "case", kind: .keyword),
+        ClassificationSpec(source: "Cons", kind: .identifier),
+        ClassificationSpec(source: "T", kind: .typeIdentifier),
+        ClassificationSpec(source: "List", kind: .typeIdentifier)
+      ]
+    )
+  }
+  
+  public func testMutatingKeyword() {
+    assertClassification(
+      "mutating func func_mutating_1() {}"
+      , expected: [
+        ClassificationSpec(source: "mutating", kind: .keyword),
+        ClassificationSpec(source: "func", kind: .keyword),
+        ClassificationSpec(source: "func_mutating_1", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testNonMutatingKeyword() {
+    assertClassification(
+      "nonmutating func func_mutating_2() {}"
+      , expected: [
+        ClassificationSpec(source: "nonmutating", kind: .keyword),
+        ClassificationSpec(source: "func", kind: .keyword),
+        ClassificationSpec(source: "func_mutating_2", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testStringLikeLiterals() {
+    assertClassification(
+      "var s1 = \"abc абвгд あいうえお\""
+      , expected: [
+        ClassificationSpec(source: "var", kind: .keyword),
+        ClassificationSpec(source: "s1", kind: .identifier),
+        ClassificationSpec(source: "\"abc абвгд あいうえお\"", kind: .stringLiteral)
+      ]
+    )
+  }
+  
+  public func testOperatorDecl() {
+    assertClassification(
+      """
+      infix operator *-* : FunnyPrecedence
+      """
+      , expected: [
+        ClassificationSpec(source: "infix", kind: .keyword),
+        ClassificationSpec(source: "operator", kind: .keyword),
+        ClassificationSpec(source: "*-*", kind: .operatorIdentifier),
+        ClassificationSpec(source: "FunnyPrecedence", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testPrecedenceGroup() {
+    XCTExpectFailure("'left' is classified as a keyword in the test")
+    assertClassification(
+      """
+      precedencegroup FunnyPrecedence {
+        associativity: left
+        higherThan: MultiplicationPrecedence
+      }
+      """
+      , expected:  [
+        ClassificationSpec(source: "precedencegroup", kind: .keyword),
+        ClassificationSpec(source: "FunnyPrecedence", kind: .identifier),
+        ClassificationSpec(source: "associativity", kind: .keyword),
+        ClassificationSpec(source: "left", kind: .identifier),
+        ClassificationSpec(source: "higherThan", kind: .keyword),
+        ClassificationSpec(source: "MultiplicationPrecedence", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testInheritence() {
+    assertClassification(
+      "protocol Prot2 : Prot {}", expected: [
+        ClassificationSpec(source: "protocol", kind: .keyword),
+        ClassificationSpec(source: "Prot2", kind: .identifier),
+        ClassificationSpec(source: "Prot", kind: .typeIdentifier)
+      ]
+    )
+  }
+  
+  public func testGenericType() throws {
+    XCTExpectFailure(" '_' is not classified")
```

Comment:
That’s also OK. Just remove the classification. I don’t really care about the classification of `_` and classifying it as a keyword never really made sense anyway.